src/flb_log.c

```diff
@@ -623,15 +618,15 @@ int flb_log_construct(struct log_message *msg, int *ret_len,
     }
     #endif // FLB_LOG_NO_CONTROL_CHARS
 
-    now = time(NULL);
```

Comment:
Make this something that can be configured through a macro that's set using CMAKE like any other features and make it default to disabled unless @edsiper vouches for the behavior change.

I saw a benchmark comparing the `cfl_time_get` and `flb_time_get` functions but didn't see one comparing `flb_time_get` with `time` which is what we need to estimate the overhead. If there is such a benchmark and I missed it please let me know so I can re-evaluate the situation.

Reply:
If we get to a point where we decide that we do not want to definitely change the granularity we need to have both versions wrapped by preprocessor conditionals where the `time` branch sets `now.tv_sec` with the appropriate value and `now.tv_nsec` to zero and the other branch just invokes `flb_time_get`.